eip712-adapter-moshi/src/main/java/pm/gnosis/eip712/adapters/moshi/TypeAdapters.kt

```diff
@@ -0,0 +1,75 @@
+package pm.gnosis.eip712.adapters.moshi
+
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.JsonQualifier
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.ToJson
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import pm.gnosis.models.Wei
+import pm.gnosis.utils.*
+import java.math.BigInteger
+
+object MoshiBuilderFactory {
+    fun makeMoshiBuilder(): Moshi.Builder {
+        return Moshi.Builder()
+            .add(WeiAdapter())
+            .add(HexNumberAdapter())
+            .add(DecimalNumberAdapter())
+            .add(DefaultNumberAdapter())
+            .add(SolidityAddressAdapter())
+    }
+}
+
+class WeiAdapter {
+    @ToJson
+    fun toJson(wei: Wei): String =
+        StringBuilder("0x").append(wei.value.toString(16)).toString()
```

Comment:
This should be the same as `wei.value.toHexStrin()`

Reply:
I had the same thought. Was just to slow to commit it already...